### PR TITLE
Fixes #1194: OEmbed http requests use "request" npm package

### LINKED ISSUE #1194: Outbound http and https requests from server to honor http_proxy env variables (oembed)
After discussion in Rocket.Chat#support and some investigation, it was concluded that links to images and pages were not embedding in the chat channel because the http_proxy environment variables were not being honored.

The solution we discussed was to make oembed detect the http_proxy variables and use them if they exist.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27593244-outbound-http-and-https-requests-from-server-to-honor-http_proxy-env-variables-oembed?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by nishimaki10**
Closes #1194

OEmbed http requests use "request" npm package instead of official node modules.

### CODE REVIEW

**Comment by engelgabriel** [comment]
Thanks @nishimaki10 
@rodrigok can you review this?

**Comment by rodrigok** [comment]
@nishimaki10 awesome, but I have some questions.

Why not use the Meteor's HTTP module http://docs.meteor.com/#/full/http_call ?

Is there some way to limit the request size? Users can past URL of big files and we don't wan't to download the entire file to get only the headers.

**Comment by nishimaki10** [comment]
@rodrigok Thanks for review!

I have used for the Meteor first time. I did not know that exists Meteor's HTTP module. Thank you for the valuable information.

But, may not be able to implement *limit the request size*. We have to control the request stream. [It's probably like this.](http://stackoverflow.com/questions/21577251/limit-body-content-length-in-a-request-in-node-js) I looking at [this](https://github.com/meteor/meteor/blob/06d64bfcd4a7677c83140324e77c608d804bad77/packages/http/httpcall_server.js), it is impossible to touch the stream.

What should I do?

**Comment by rodrigok** [comment]
@nishimaki10 If you can limit the request using the lib `request` go ahead and use it, but use the internal one `HTTPInternals.NpmModules.request.module`. What do you think?

**Comment by nishimaki10** [comment]
@rodrigok I see. I will try to implement using the `HTTPInternals.NpmModules.request.module`. Thanks!

File: packages/rocketchat-oembed/server/server.coffee [hunk 1]
```diff
@@ -13,73 +8,50 @@ getUrlContent = (urlObj, redirectCount = 5, callback) ->
...
+		headers = response.headers
+
+	stream.on 'data', (chunk) ->
+		chunks.push chunk
+		chunksTotalLength += chunk.length
+		if chunksTotalLength > 250000
+			stream.abort()
+
+	stream.on 'end', Meteor.bindEnvironment ->
+		buffer = Buffer.concat(chunks)
```
Comment (rodrigok) [inline packages/rocketchat-oembed/server/server.coffee]: What about the gunzip? The library do the unzip for each chunk? I don't think so, can you test?

Comment (nishimaki10) [inline packages/rocketchat-oembed/server/server.coffee]: Yes. The request module pipe gunzip for each chunk. [see also](https://github.com/request/request/blob/master/request.js#L924). I think that this library is made well.
My Test:
[gziped site](http://checkgzipcompression.com/?url=http%3A%2F%2Fcheckgzipcompression.com%2F)
Screenshot:
screenshot-localhost 3000 2016-03-23 07-04-02

**Comment by nishimaki10** [comment]
Twitter, GitHub, etc, etc... I tested many sites. All OK.
The `content-encoding: gzip` header exists properly in response if site enabled gzip. e.g. Twitter, GitHub.
Thanks.

**Comment by rodrigok** [comment]
LGTM

**Comment by rodrigok** [comment]
@nishimaki10 sorry about that, but, can you fix the conflicts?

**Comment by nishimaki10** [comment]
@rodrigok OK, solved.

**Pull request merged**